src/index.d.ts

```diff
@@ -0,0 +1,280 @@
+import { ReactElement, ComponentType } from 'react'
+import { ReactTestInstance, ReactTestRenderer, act } from 'react-test-renderer'
+
+// EVENTS
+// ------
+
+// TODO: type second parameter
+export declare class NativeEvent {
+  constructor(typeArg: string, event?: any)
+}
+
+export declare function getEventHandlerName(key: string): string
+
+export interface FireEventFn {
+  (element: FiberRoot, event: NativeEvent): any
+  focus(element: FiberRoot, init?: any): any
+  blur(element: FiberRoot, init?: any): any
+  change(element: FiberRoot, init?: any): any
+  changeText(element: FiberRoot, value: string): any
+  contentSizeChange(element: FiberRoot, init?: any): any
+  endEditing(element: FiberRoot, init?: any): any
+  keyPress(element: FiberRoot, init?: any): any
+  submitEditing(element: FiberRoot, init?: any): any
+  layout(element: FiberRoot, init?: any): any
+  selectionChange(element: FiberRoot, init?: any): any
+  longPress(element: FiberRoot, init?: any): any
+  press(element: FiberRoot, init?: any): any
+  pressIn(element: FiberRoot, init?: any): any
+  pressOut(element: FiberRoot, init?: any): any
+  momentumScrollBegin(element: FiberRoot, init?: any): any
+  momentumScrollEnd(element: FiberRoot, init?: any): any
+  scroll(element: FiberRoot, init?: any): any
+  scrollBeginDrag(element: FiberRoot, init?: any): any
+  scrollEndDrag(element: FiberRoot, init?: any): any
+  load(element: FiberRoot, init?: any): any
+  error(element: FiberRoot, init?: any): any
+  progress(element: FiberRoot, init?: any): any
+}
+export declare const fireEvent: FireEventFn
+
+// GET NODE TEXT
+// -------------
+
+export declare function getNodeText(node: FiberRoot): string
+
+// GET QUERIES FOR ELEMENT
+// -----------------------
+
+export declare function getQueriesForElement<T = Queries>(element: ReactElement, queries?: T): BoundQueries<T>
+export declare function within<T = Queries>(element: ReactElement, queries?: T): BoundQueries<T>
+
+// PREETY PRINT
+// ------------
+
+// TODO: options are pretty-format options
+export declare function prettyPrint(element: ReactTestRenderer | FiberRoot, maxLength?: number, options?: any): string
+
+// QUERIES
+// -------
+
+type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>
+type Bound<T> = T extends (arg: any, ...rest: infer U) => infer V ? (...args: U) => V : never
+type BoundQueries<T> = { [P in keyof T]: Bound<T[P]> }
+
+export type FiberRoot = Omit<ReactTestInstance, 'find' | 'findAllByProps' | 'findAllByType' | 'findByProps' | 'findByType' | 'instance'>
```

Comment:
Can you talk through your thinking in naming this `FiberRoot`? I was thinking something along the lines of `NativeInstance` or `NativeElement` although neither of those are quite right, either.

Reply:
I am open to changing this. The `FiberRoot` was chosen to make the types consistent with documentation https://www.native-testing-library.com/docs/api-queries#bya11yhint

Reply:
This brings up a good point, this change will likely require updating the docs also so that the types match.